infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LRRollingUpgradeHandler.java

```diff
@@ -129,6 +129,10 @@ public boolean isLRUpgradeInProgress(TxnContext txnContext) {
      * @param txnContext All of the above must execute in the same transaction passed in.
      */
     public void migrateData(TxnContext txnContext) {
+        if(LogReplicationMetadataManager.isMigrationComplete(txnContext)) {
+            log.info("Data migration for LR V2 has already been completed.");
```

Comment:
as Shreay mentioned below - don't we want to set the in-memory field `isClusterAllAtV2` to true here? This will avoid having to do a `getTable()` right?

Reply:
the variable was getting set in `isLRUpgradeInProgress()`.

I have anyways changed the logic completely now :) 
Have changed it to check the type of record in status and event table..